Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,70 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
-        puts
-        puts "External commands"
-        puts Formatter.columns(exts)
+    # TODO: Goal is to not require this function at all.
+    command = CommandsCommand.new
+    command.check_for_errors
```

Comment:
Actually, the `ARGV` is equal to: 
```["-I", "/usr/local/Homebrew/Library/Homebrew/test", "--color", "--require", "spec_helper", "--format", "progress", "--format", "ParallelTests::RSpec::RuntimeLogger", "--out", "/Users/razahussain/Library/Caches/Homebrew/tests/parallel_runtime_rspec.log", "--color", "--tty", "test/cmd/commands_spec.rb"]```, when `described_class.new ` is called in the test cases.